gen/struct_test.go

```diff
@@ -979,10 +994,45 @@ func TestStructJSON(t *testing.T) {
 		v := reflect.New(reflect.TypeOf(tt.v).Elem()).Interface()
 		if assert.NoError(t, json.Unmarshal([]byte(tt.j), v), "failed to decode %q", tt.j) {
 			assert.Equal(t, tt.v, v)
+			assert.Equal(t, tt.v, v)
 		}
 	}
 }
 
+func TestJSONOmitBehaviour(t *testing.T) {
+	omit := ts.Omit{Serialized: "foo", Hidden: "bar"}
+	b, err := json.Marshal(&omit)
+
+	assert.NoError(t, err, "should marshal")
+	assert.Equal(t, b, []byte(`{"serialized":"foo"}`))
+
+	omit2 := ts.Omit{}
+	err = json.Unmarshal([]byte(`{"serialized":"foo","hidden":"bar"}`), &omit2)
+
+	assert.NoError(t, err, "should unmarshal")
+	assert.Equal(t, omit2.Serialized, "foo")
+	assert.Equal(t, omit2.Hidden, "")
+}
+
+func TestStructGoTags(t *testing.T) {
+	gt := &ts.GoTags{
+		Foo:    "shouldomit",
+		FooBar: "shouldnotomit",
+	}
+	b, err := json.Marshal(gt)
+	assert.NoError(t, err, "should marshal")
+	assert.Equal(t, b, []byte(`{"foobar":"shouldnotomit"}`))
+
+	foo, _ := reflect.TypeOf(gt).Elem().FieldByName("Foo")
+	assert.Equal(t, "json:\"-\" foo:\"bar\"", string(foo.Tag))
```

Comment:
You can use the ` string literal form to avoid escaping

    `json:"-" foo:"bar"`